src/module_ncio.f90

```diff
@@ -1,81 +1,58 @@
 !> @file
-!! Module for reading/writing netcdf files such as the lat/lon grid history files output by the GFS.
-!! writing requires a template file.
+!! @brief Module for reading/writing netcdf files such as the lat/lon grid history files output by the GFS..
+!! @author jeff whitaker <jeffrey.s.whitaker@noaa.gov> @date 201910
+
+!> writing requires a template file.
 !! Handles 32 and 64 bit real variables, 8, 16 and 32 bit integer
 !! variables and char variables. Variables can have up to 5 dimensions.
 !! @author jeff whitaker <jeffrey.s.whitaker@noaa.gov> @date 201910
 module module_ncio
-!
+
   use netcdf
   use mpi
-!
+  
   implicit none
   private
-!
+
   type Variable
-! netCDF variable ID
-     integer varid 
-! number of dimensions
-     integer ndims 
-! netCDF data type
-     integer dtype 
-! number of attributes
-     integer natts 
-! compression level (if > 0)
-     integer deflate_level 
-! shuffle filter?
-     logical shuffle  
-! has an unlimited dim?
-     logical hasunlim 
-! variable name
-     character(len=nf90_max_name) :: name 
-! netCDF dimension IDs
-     integer, allocatable, dimension(:) :: dimids 
-!
-! indices into Dataset%dimensions for associated dimensions.
-     integer, allocatable, dimension(:) :: dimindxs
-! names of associated dimensions.
-     character(len=nf90_max_name), allocatable, dimension(:) :: dimnames
-! current dimension lengths (updated after every write_vardata call)
+     integer varid !< NetCDF variable ID.
+     integer ndims !< Number of dimensions.
+     integer dtype !< netCDF data type
+     integer natts !< number of attributes
+     integer deflate_level !< compression level (if > 0)
+     logical shuffle  !< shuffle filter?
+     logical hasunlim !< has an unlimited dim?
+     character(len=nf90_max_name) :: name !< variable name
+     integer, allocatable, dimension(:) :: dimids !< netCDF dimension IDs
+
+     integer, allocatable, dimension(:) :: dimindxs !< indices into Dataset%dimensions for associated dimensions.
+     character(len=nf90_max_name), allocatable, dimension(:) :: dimnames !< names of associated dimensions.
+     !< current dimension lengths (updated after every write_vardata call)
      integer, allocatable, dimension(:) :: dimlens
      integer, allocatable, dimension(:) :: chunksizes
   end type Variable
-!
+
   type Dimension
-! netCDF dimension ID
-     integer dimid 
-! dimension length (updated after every write_vardata call)
-     integer len 
-! unlimited?
-     logical isunlimited
-! name of dimension
-     character(len=nf90_max_name) :: name 
+     integer dimid !< netCDF dimension ID
+     integer len   !< dimension length (updated after every write_vardata call)
+     logical isunlimited !< unlimited? 
+     character(len=nf90_max_name) :: !< name name of dimension
```

Comment:
Your comment marker is in the wrong spot, resulting in the loss of the name variable.

You should be compiling before each commit, so that this does not happen. Also you should be building with -DENABLE_DOCS=ON, and looking at the doxygen warnings. We want zero warnings.